packages/data-transport-layer/src/services/l1-ingestion/service.ts

```diff
@@ -453,12 +453,18 @@ export class L1IngestionService extends BaseService<L1IngestionServiceOptions> {
 
   private async _findStartingL1BlockNumber(): Promise<number> {
     const currentL1Block = await this.state.l1RpcProvider.getBlockNumber()
+    const filter =
+      this.state.contracts.Lib_AddressManager.filters.OwnershipTransferred()
+
+    for (let i = 0; i < currentL1Block; i += 2000) {
+      const start = i
+      const end = Math.min(i + 2000, currentL1Block)
```

Comment:
Isn't `end` in `queryFilter` inclusive? So this could include the end blocks twice?

Reply:
I think that's correct, although I don't think it matters in this case (because we just care about finding the first event, doesn't matter if we scan the same block twice).